nservicebus/azure-service-bus/retries.md

```diff
@@ -5,36 +5,49 @@ tags:
+
+To implement Immediate Retries, the native Azure Service Bus `MaxDeliveryCount` property of the `BrokeredMessage` is used to track the processing attempts by the endpoint instances for a specific message. If the number of attempts exceeds the `MaxRetries` setting, then the message will be forwarded to the delayed retries if configured, and eventually to at the error queue in case error persists.
+
+A scaled out endpoint will retry the message by any of its instances, before one of them decides to send the message to the delayed retries or the error queue. So in effect, the number of immediate retries attempts will never exceed the defined number of immediate retries for an endpoint.
+
+Versions 6 and below the default value for `MaxDeliveryCount` is 6. Versions 7 and higher it has been changed to be equal to a number of immediate retries for an endpoint plus one. For system queues, such as audit and error queues, the default was changed to 10. Should endpoint disable immediate retries, immediate retries will be set to 10.
+
+
+### Keeping MaxDeliveryCount in sync with immediate retries
+
+Whenever an endpoint is redeployed with a new Immediate Retries value, endpoint specific queues, such as endpoint`s input queue, will be updated with the new value.
+
+
+### Modifying system queues MaxDeliveryCount
+
+A newly deployed or a re-deployed endpoint will not modify system queues `MaxDeliveryCount` set by end users.
 
-In order to implement Immediate Retries, NServiceBus maintains an internal counter to track the processing attempts by the endpoint instance for a specific message. If the number of attempts exceeds the `MaxRetries` setting, then the message will be forwarded to the error queue.
 
-Each endpoint's instance maintains its own internal counter for retries. If an endpoint is scaled out then the message can be retried a few times by each instance, before one of them decides to send the message to the error queue. So in effect the global retry count, as seen by the Azure Service Bus entity, will in the worst case scenario be equal to `MaxRetries * number of instances`.
+### Excessive prefetching and MaxDeliveryCount 
 
-In Azure Service Bus Versions 6 and below the default value for `MaxDeliveryCount` is 6. In Azure Service Bus Versions 7 and higher it has been changed to 10, based on the assumption that by default there are 2 instances of each endpoint and `MaxRetries` has the default value of 5.
+When messages are received, client receivers prefetch messages. Endpoint taking a long time to process a message will cause prefetched messages to lose lock tokens. The broker will consider that as a failed processing attempt increase `DeliveryCount` counter on the messages w/o processing attempt.  
 
+This can cause a message to be sent to the Delayed Retries or the error queue without being retried the desired number of times. The problem can be addressed by reducing the `PrefetchCount` on message receivers and retrying the failed messages.
```

Comment:
I'm not clear if the problem is excessive prefetching or slow processing. If it's the first then it might be clearer, if we replace

> Endpoint taking a long time to process a message will cause prefetched messages to lose lock tokens.

with something like: 

NSB uses prefetching in order to ... Messages are prefetched in batches... If it takes a long time to process a (prefetched) batch, then ... To prevent the issue lower the prefetch count.